rdf/common/src/main/scala/org/w3/banana/io/AnyRDFReader.scala

```diff
@@ -0,0 +1,12 @@
+package org.w3.banana
+package io
+
+
+/** RDF reader. */
+trait AnyRDFReader[Rdf <: RDF, M[_]] {
+  
+	/** Read triples from the given location. The syntax is determined from input source URI
+   *  (content negotiation or extension). */
+   def readAnyRDFSyntax( uri:String) : M[Rdf#Graph]
```

Comment:
Ah, I just realized that we were talking about different things. I read "separate package" from @bblfish's comment and understood it as "separate module". IMO, under `https://github.com/w3c/banana-rdf/tree/series/0.8.x/rdf/common/src/main/scala/org/w3/banana/io` is just fine for the abstractions. The implementations would live in jena and sesame, as expected. So as far as I am concerned, you're already fine.